src/main/java/org/elasticsearch/index/query/AndQueryParser.java

```diff
@@ -36,6 +36,8 @@
 @Deprecated
 public class AndQueryParser extends BaseQueryParserTemp {
 
+    private static final AndQueryBuilder AND_QUERY_BUILDER = new AndQueryBuilder();
```

Comment:
I was wondering if we should use some convention on naming, like `INSTANCE` or `PROTOTYPE`, I have no strong preference though